src/index.ts

```diff
@@ -9,6 +9,7 @@ type PersistOptions = {
   selector?: string
   keyPrefix?: string
   storage?: Pick<Storage, 'getItem' | 'setItem'>
+  fields?: NodeList | Node[]
 }
```

Comment:
it could be helpful to use a discriminating union so that you can't define both `fields` and `selector` in the same `PersistOptions`

```ts
type PersistOptionsWithSelector = {
   selector: string
   fields?: never
 }

type PersistOptionsWithFields = {
   fields: NodeList | Node[]
   selector?: never
 }

type PersistOptions = (PersistOptionsWithFields | PersistOptionsWithSelector) & {
   keyPrefix?: string
   storage?: Pick<Storage, 'getItem' | 'setItem'>
}
```

👉  [I put the above in a TS Playground for you to play with.](https://www.typescriptlang.org/play?#code/C4TwDgpgBAChBOBnAlo4B5MxkHsB2iA6ssABYDKEANhAMbA7xQC8UA3gFBTeLV0PwAXFDTxkeAOZduAM2TUAJogD8wvBABuCLgF8OHUJFgIUaTNnxESpAGLyqSlu2lQ5ixMIByOBRAAyqMBQAD5Q3r4A2gC6Lrw09IyqUOpa8Lr6htBwSIHmuAROABTZphhY+VZkdu4hxjlm5ZbEZJTxAgCUUABkztxQANYQIDDwEHIAHkmi4lJ9aIwAhhIQSTDItP0APOQCSxAANFAA5MvAAJLAEAC2R7VHvOeXNwB8HHocMgCuePT5UJD1YAAJQgiE+VwWACMaNUHIhCsgFMJppJDjhGgRhCVchjEJ1OH0oAB6IlQAB0FIgwFob30ANKILBEOhEFhSkK9z4CXgR0ObDi-EYwgARMLDoNhqMJsIjhKjjp2nSTIFGeCoTD7Oyjm44bz2DqlMJouKhiMxshxjK5QqlYDVcyNe4OSkEHq2AaPFBjSIuQIRWKBqapRarUN5Yr9FAgA)